File: tests/imperative/arrays/test00.ml

#use "tests/imperative/arrays/__arrays.ml"

let d = ref lD {"f" = 1} in
let _ :: {(= v 1)} = ([;lD] impGetProp) (d, "f") in

let a = ref lA <0, 1> in
let _ :: {(= v 2)} = ([Top;lA] impGetProp) (a, "length") in
let _ :: {(= v 0)} = ([Top;lA] impGetArr) (a, 0) in
let _ :: {(= v 1)} = ([Top;lA] impGetArr) (a, 1) in
let _ :: {(= v undefined)} = ([Top;lA] impGetArr) (a, 2) in

0

File: tests/djsLite/arrays/test00.js

var a = /*: lA */ [0,"hi",true];

/*: {(= v 0)} */ (/*: [Top;lA;] */ a[0]);

/*: {(= v "hi")} */ (/*: [Top;lA;] */ a[1]);

/*: {(= v True)} */ (/*: [Top;lA;] */ a[2]);

/*: {(= v undefined)} */ (/*: [Top;lA;] */ a[3]);

/*: {(= v 3)} */ (/*: [Top;lA;] */ a["length"]);

/*: {(= v 3)} */ (/*: [Top;lA;] */ a.length);

var s = "length";

/*: {(= v 3)} */ (/*: [Top;lA;] */ a[s]);

var i = 1;

/*: {(= v "hi")} */ (/*: [Top;lA;] */ a[i]);
